Yep
```
        arrow_table = pa.Table.from_arrays([sparse_union_array], schema=pa.schema([("data", sparse_union_array.type)]))
>       duckdb_cursor.register('invalid_union', arrow_table)
E       duckdb.duckdb.InternalException: INTERNAL Error: Assertion triggered in file "/Users/thijs/DuckDBLabs/duckdb/src/common/types.cpp" on line 1343: !members.empty()
```

--- a/src/function/table/arrow.cpp
+++ b/src/function/table/arrow.cpp
@@ -148,6 +148,9 @@ static unique_ptr<ArrowType> GetArrowLogicalTypeNoDictionary(ArrowSchema &schema
 
 		child_list_t<LogicalType> members;
 		vector<unique_ptr<ArrowType>> children;
+		if (schema.n_children == 0) {
+			throw InvalidInputException("Attempted to convert a UNION with no fields to DuckDB which is not supported");
+		}
 		for (idx_t type_idx = 0; type_idx < (idx_t)schema.n_children; type_idx++) {
 			auto type = schema.children[type_idx];
 

--- a/tools/pythonpkg/tests/fast/arrow/test_arrow_types.py
+++ b/tools/pythonpkg/tests/fast/arrow/test_arrow_types.py
@@ -30,3 +30,18 @@ class TestArrowTypes(object):
             match='Attempted to convert a STRUCT with no fields to DuckDB which is not supported',
         ):
             duckdb_cursor.register('invalid_struct', arrow_table)
+
+    def test_invalid_union(self, duckdb_cursor):
+        # Create a sparse union array from dense arrays
+        types = pa.array([0, 1, 1], type=pa.int8())
+        sparse_union_array = pa.UnionArray.from_sparse(types, [], type_codes=[])
+
+        arrow_table = pa.Table.from_arrays([sparse_union_array], schema=pa.schema([("data", sparse_union_array.type)]))
+        with pytest.raises(
+            duckdb.InvalidInputException,
+            match='Attempted to convert a UNION with no fields to DuckDB which is not supported',
+        ):
+            duckdb_cursor.register('invalid_union', arrow_table)
+
+            res = duckdb_cursor.sql("select * from invalid_union").fetchall()
+            print(res)

